neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -0,0 +1,151 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+"""
+See class comment for details
+"""
+from typing import Any
+from typing import Dict
+from typing import List
+
+from neuro_san.interfaces.concierge_session import ConciergeSession
+from neuro_san.internals.network_providers.agent_network_storage import AgentNetworkStorage
+from neuro_san.service.http.interfaces.agent_authorizer import AgentAuthorizer
+from neuro_san.service.generic.async_agent_service import AsyncAgentService
+from neuro_san.service.generic.async_agent_service_provider import AsyncAgentServiceProvider
+from neuro_san.session.direct_concierge_session import DirectConciergeSession
+from neuro_san.service.mcp.util.mcp_errors_util import MCPErrorsUtil
+from neuro_san.service.http.logging.http_logger import HttpLogger
+
+
+class MCPToolsProcessor:
+    """
+    Class implementing "tools"-related MCP requests.
+    """
+
+    def __init__(self,
+                 logger: HttpLogger,
+                 network_storage_dict: AgentNetworkStorage,
+                 agent_policy: AgentAuthorizer):
+        self.logger: HttpLogger = logger
+        self.network_storage_dict: AgentNetworkStorage = network_storage_dict
+        self.agent_policy: AgentAuthorizer = agent_policy
+
+    async def list_tools(self, request_id, metadata: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        List available MCP tools.
+        :param request_id: MCP request id;
+        :param metadata: http-level request metadata;
+        :return: json dictionary with tools list in MCP format
+        """
+        public_storage: AgentNetworkStorage = self.network_storage_dict.get("public")
+        data: Dict[str, Any] = {}
+        session: ConciergeSession = DirectConciergeSession(public_storage, metadata=metadata)
+        result_dict: Dict[str, Any] = session.list(data)
+        tools_description: List[Dict[str, Any]] = []
```

Comment:
It would make sense to expose an mcp switch in the default dictionary that is returned for each entry in  ManifestDictConfigFilter for greater flexibility of mixing/matching than just what the Concierge service returns.

It might be the case that:
* not every public network wants to be mcp exposed
* maybe some protected networks want to be mcp exposed.

That said it might make sense to have mcp exposure on by default for ease of use.